sklearn/utils/extmath.py

```diff
@@ -210,28 +219,45 @@ def randomized_range_finder(A, size, n_iter, random_state=None):
     Finding structure with randomness: Stochastic algorithms for constructing
     approximate matrix decompositions
     Halko, et al., 2009 (arXiv:909) http://arxiv.org/pdf/0909.4061
+
+    An implementation of a randomized algorithm for principal component
+    analysis
+    A. Szlam et al. 2014
     """
     random_state = check_random_state(random_state)
 
-    # generating random gaussian vectors r with shape: (A.shape[1], size)
-    R = random_state.normal(size=(A.shape[1], size))
-
-    # sampling the range of A using by linear projection of r
-    Y = safe_sparse_dot(A, R)
-    del R
-
-    # perform power iterations with Y to further 'imprint' the top
-    # singular vectors of A in Y
-    for i in xrange(n_iter):
-        Y = safe_sparse_dot(A, safe_sparse_dot(A.T, Y))
-
-    # extracting an orthonormal basis of the A range samples
-    Q, R = linalg.qr(Y, mode='economic')
+    # Generating uniform random vectors r with shape: (A.shape[1], size)
+    Q = random_state.normal(size=(A.shape[1], size))
+
+    # Deal with "auto" mode
+    if power_iteration_normalizer is 'auto':
+        if n_iter <= 2:
+            power_iteration_normalizer = 'none'
+        else:
+            power_iteration_normalizer = 'LU'
+
+    # Perform power iterations with Q to further 'imprint' the top
+    # singular vectors of A in Q
+    for i in range(n_iter):
+        if power_iteration_normalizer is 'none':
```

Comment:
string comparison should be made with `==` otherwise it depends on the interning state of the string object.